app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/CreateDBTablePageSolution.java

```diff
@@ -0,0 +1,436 @@
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                List<NewPage> pageList = applicationJson.getPageList();
+
+                if (pageList.isEmpty()) {
+                    return Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, TEMPLATE_APPLICATION_FILE));
+                }
+
+                //TODO for MVP we are supporting single page with only one layout
+                Layout layout = pageList.get(0).getUnpublishedPage().getLayouts().get(0);
+                layout.setId(null);
+
+                //We are expecting only one table will be present for MVP
+                Table templateTable = applicationJson.getDatasourceList().get(0).getStructure().getTables().get(0);
+
+                mappedColumnsAndTableName.putAll(mapTableColumnNames(templateTable, table));
+                mappedColumnsAndTableName.put(TEMPLATE_TABLE_NAME, tableName);
+
+                Set<String> deletedWidgets = new HashSet<>();
+                layout.setDsl(
+                    extractAndUpdateAllWidgetFromDSL(layout.getDsl(), mappedColumnsAndTableName, deletedWidgets)
+                );
+                return layoutActionService.updateLayout(savedPageId.get(), layoutId, layout)
+                    .then(Mono.zip(
+                        Mono.just(datasource),
+                        Mono.just(applicationJson.getActionList()),
+                        Mono.just(deletedWidgets)
+                    ));
+            })
+            .flatMap(tuple -> {
+
+                Datasource datasource = tuple.getT1();
+                List<NewAction> templateActionList = tuple.getT2();
+                Set<String> deletedWidgets = tuple.getT3();
+                return cloneActions(datasource, tableName, savedPageId.get(), templateActionList, mappedColumnsAndTableName, deletedWidgets)
+                    .flatMap(actionDTO -> StringUtils.equals(actionDTO.getName(), SELECT_QUERY) ?
+                        layoutActionService.setExecuteOnLoad(actionDTO.getId(), true) : Mono.just(actionDTO))
+                    .then(applicationPageService.getPage(savedPageId.get(), false));
+            });
+    }
+
+    private Mono<NewPage> getPage(String applicationId, String pageId, String tableName) {
+
+        if(pageId != null) {
+            return newPageService.findById(pageId, AclPermission.MANAGE_PAGES)
+                .switchIfEmpty(Mono.error(new AppsmithException(
+                    AppsmithError.ACL_NO_RESOURCE_FOUND, FieldName.PAGE, pageId))
+                );
+        }
+
+        return newPageService.findByApplicationId(applicationId, AclPermission.MANAGE_PAGES, false)
+            .collectList()
+            .flatMap(pages -> {
+                String pageName = "Admin Page: " + tableName;
+                long maxCount = 0l;
+                for (PageDTO pageDTO : pages) {
+                    if (pageDTO.getName().matches("^" + pageName + ".*")) {
```

Comment:
If we are creating a page for the table `products`, we are trying to name the page as `Products`, correct? So we are checking if a page by that name already exists and adding a numerical suffix, if it does.

If my understanding above is correct, then this regex is a bit too lax. It'll match the page named `Products Main` also, which we don't want. Something like the below might work better?

```suggestion
                    if (pageDTO.getName().matches("^" + Pattern.quote(pageName) + "\d*$")) {
```